File: src/Application Framework/Entities/Picklists/PickList.vb
Namespace Entity

    Namespace PickLists

        Public Class PickList

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "PickList Properties"

            Private _ManagerID As Integer = 0
            Public ReadOnly Property ManagerID() As Integer
                Get
                    Return _ManagerID
                End Get
            End Property
            Public WriteOnly Property SetManagerID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ManagerID", Value)
                End Set
            End Property

            Private _EnumTypeID As Integer = 0
            Public ReadOnly Property EnumTypeID() As Integer
                Get
                    Return _EnumTypeID
                End Get
            End Property
            Public WriteOnly Property SetEnumTypeID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_EnumTypeID", Value)
                End Set
            End Property

            Private _Name As String = String.Empty
            Public ReadOnly Property Name() As String
                Get
                    Return _Name
                End Get
            End Property
            Public WriteOnly Property SetName() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Name", Value)
                End Set
            End Property

            Private _Description As String = String.Empty
            Public ReadOnly Property Description() As String
                Get
                    Return _Description
                End Get
            End Property
            Public WriteOnly Property SetDescription() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Description", Value)
                End Set
            End Property

            Private _PercentageRate As Double = 0.0
            Public ReadOnly Property PercentageRate() As Double
                Get
                    Return _PercentageRate
                End Get
            End Property
            Public WriteOnly Property SetPercentageRate() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PercentageRate", Value)
                End Set
            End Property

            Private _Mandatory As Boolean = False
            Public ReadOnly Property Mandatory() As Boolean
                Get
                    Return _Mandatory
                End Get
            End Property
            Public WriteOnly Property SetMandatory() As Boolean
                Set(ByVal Value As Boolean)
                    _Mandatory = Value
                End Set
            End Property
#End Region

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Framework/Entities/Picklists/PickList.cs
using System.Collections;

namespace FSM.Entity
{
    namespace PickLists
    {
        public class PickList
        {
            private DataTypeValidator _dataTypeValidator;

            public PickList()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            

            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            
            private int _ManagerID = 0;

            public int ManagerID
            {
                get
                {
                    return _ManagerID;
                }
            }

            public object SetManagerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ManagerID", value);
                }
            }

            private int _EnumTypeID = 0;

            public int EnumTypeID
            {
                get
                {
                    return _EnumTypeID;
                }
            }

            public object SetEnumTypeID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EnumTypeID", value);
                }
            }

            private string _Name = string.Empty;

            public string Name
            {
                get
                {
                    return _Name;
                }
            }

            public object SetName
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Name", value);
                }
            }

            private string _Description = string.Empty;

            public string Description
            {
                get
                {
                    return _Description;
                }
            }

            public object SetDescription
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Description", value);
                }
            }

            private double _PercentageRate = 0.0;

            public double PercentageRate
            {
                get
                {
                    return _PercentageRate;
                }
            }

            public object SetPercentageRate
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PercentageRate", value);
                }
            }

            private bool _Mandatory = false;

            public bool Mandatory
            {
                get
                {
                    return _Mandatory;
                }
            }

            public bool SetMandatory
            {
                set
                {
                    _Mandatory = value;
                }
            }

            
        }
    }
}